index.bs

```diff
@@ -1851,6 +1851,7 @@ partial interface MLGraphBuilder {
     1. If any element in |options|.{{MLConv2dOptions/strides}} is equal to 0, then [=exception/throw=] a {{TypeError}}.
     1. If |options|.{{MLConv2dOptions/dilations}} does not [=map/exist=], set it to the [=/list=] « 1, 1 ».
     1. Otherwise, if |options|.{{MLConv2dOptions/dilations}}'s [=list/size=] is not 2, then [=exception/throw=] a {{TypeError}}.
+    1. If any element in |options|.{{MLConv2dOptions/dilations}} is equal to 0, then [=exception/throw=] a {{TypeError}}.
```

Comment:
I was going to question this one as potentially overvalidation, since DML accepts 0 dilations just fine (they are mathematically sound), but then TF rejects it. So yeah, reasonable to keep.

## DML:
```
dxdispatch.exe models\dml-convolution-zero-dilations.json
```
```
{
    "$schema": "./_schema.json",
    
    "resources": 
    {
        "input": 
        {
            "initialValuesDataType": "FLOAT32",
            "initialValues": { "valueCount": 9, "valueStart": 1, "valueDelta": 1 }
        },
        "filter": 
        {
            "initialValuesDataType": "FLOAT32",
            "initialValues": [0.5, 0.5, 0.5, 0.5]
        },
        "output": 
        {
            "initialValuesDataType": "FLOAT32",
            "initialValues": { "valueCount": 4, "value": 0 }
        }
    },

    "dispatchables": 
    {
        "conv2d": 
        {
            "type": "DML_OPERATOR_CONVOLUTION",
            "desc": 
            {
                "InputTensor": { "DataType": "FLOAT32", "Sizes": [1,1,3,3] },
                "FilterTensor": { "DataType": "FLOAT32", "Sizes": [1,1,2,2] },
                "OutputTensor": { "DataType": "FLOAT32", "Sizes": [1,1,2,2] },
                "Mode": "DML_CONVOLUTION_MODE_CROSS_CORRELATION",
                "Direction": "DML_CONVOLUTION_DIRECTION_FORWARD",
                "DimensionCount": 2,
                "Strides": [1,1],
                "Dilations": [0,0],
                "StartPadding": [0,0],
                "EndPadding": [0,0],
                "OutputPadding": [0,0],
                "GroupCount": 1
            }
        }
    },

    "commands": 
    [
        {
            "type": "dispatch",
            "dispatchable": "conv2d",
            "bindings":
            {
                "InputTensor": "input",
                "FilterTensor": "filter",
                "OutputTensor": "output"
            }
        },
        { "type": "print", "resource": "output" }
    ]
}
```
```
Running on 'NVIDIA GeForce RTX 2080'
Dispatch 'conv2d': 1 iterations, 0.7460 ms median (CPU), 0.007744 ms median (GPU)
Resource 'output': 6, 8, 12, 14
```

## TensorFlow:
```
import tensorflow as tf

output = tf.nn.conv2d(
    tf.random.normal([1,1,10,10]), 
    tf.random.normal([1,1,10,10]), 
    strides=[1,1],
    dilations=[0,0],
    padding='SAME',
)
print(output)
```
```
INVALID_ARGUMENT: Dilated rates should be larger than 0.
```

Reply:
PyTorch also rejects the 0 dilations

```
import torch
m = torch.nn.Conv2d(16, 33, 3, stride=2, dilation=0)
input = torch.randn(20, 16, 50, 100)
output = m(input)
```

```
RuntimeError: non-positive dilation is not supported
```